### PR TITLE
fix: fix unexpected error when provider and consumer exist together

### PR DESCRIPTION
**Pull request by zhaoyunxing92**
**What this PR does**: 
解决consumer和provider同时存在的时候启动异常
**Which issue(s) this PR fixes**: 
Fixes #
https://github.com/apache/dubbo-go/issues/1764
**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [x] All ut passed (run 'go test ./...' in project root)
- [x] After go-fmt ed , run 'go fmt project' using goland.
- [x] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [x] After import formatted, (using [imports-formatter](https://github.com/dubbogo/tools#5-how-to-get-imports-formatter) to run 'imports-formatter .' in project root, to format your import blocks, mentioned in [CONTRIBUTING.md](https://github.com/apache/dubbo-go/blob/master/CONTRIBUTING.md) above) 
- [x] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [x] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

### CODE REVIEW

File: config/consumer_config.go [hunk 1]
```diff
@@ -137,24 +137,24 @@ func (cc *ConsumerConfig) Load() {
 	var count int
 	for {
 		checkok := true
-		for _, refconfig := range cc.References {
-			if (refconfig.Check != nil && *refconfig.Check) ||
-				(refconfig.Check == nil && cc.Check) ||
-				(refconfig.Check == nil) { // default to true
+		for key, ref := range cc.References {
+			if (ref.Check != nil && *ref.Check && GetProviderService(key) == nil) ||
+				(ref.Check == nil && cc.Check && GetProviderService(key) == nil) ||
```
Comment (Mulavar) [inline config/consumer_config.go:142]: 一般references的key和services的key一般不一样吧，那使用references的key能找到对应的provider service么？

File: config/consumer_config.go [hunk 2]
```diff
@@ -137,24 +137,24 @@ func (cc *ConsumerConfig) Load() {
 	var count int
 	for {
 		checkok := true
-		for _, refconfig := range cc.References {
-			if (refconfig.Check != nil && *refconfig.Check) ||
-				(refconfig.Check == nil && cc.Check) ||
-				(refconfig.Check == nil) { // default to true
+		for key, ref := range cc.References {
+			if (ref.Check != nil && *ref.Check && GetProviderService(key) == nil) ||
```
Comment (ChangedenCZD) [inline config/consumer_config.go:141]: 这些引用提取为变量会不会整洁很多

Comment (zhaoyunxing92) [reply config/consumer_config.go:141]: 很好的提议，应该去解决check为啥是指针类型的

File: config/consumer_config.go (hunk 1 above)
Comment (zhaoyunxing92) [reply config/consumer_config.go:142]: 不保证用户会配置错

**Review by LaurenceLiZhixin — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by baerwang — APPROVED** [review]

**Pull request merged**